src/lib/y2network/ip_address.rb

```diff
@@ -0,0 +1,97 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "ipaddr"
+require "forwardable"
+
+module Y2Network
+  # This class represents an IP address
+  #
+  # The IPAddr from the Ruby standard library drops the host part according to the netmask. The
+  # problem is that YaST uses a CIDR-like string, including the host part, to set IPADDR in ifcfg-*
+  # files (see man 5 ifcfg for further details).
+  #
+  # @example IPAddr from standard library behavior
+  #  ip = IPAddr.new("192.168.122.1/24")
+  #  ip.to_s #=> "192.168.122.0/24"
+  #
+  # However, what we need is to be able to keep the host part
+  #
+  # @example IPAddress behaviour
+  #   ip = IPAddress.new("192.168.122.1/24")
+  #   ip.to_s #=> "192.168.122.1/24"
+  class IPAddress
+    extend Forwardable
+
+    # @return [IPAddr] IP address
+    attr_reader :address
+    # @return [Integer] Prefix
+    attr_reader :prefix
+
+    def_delegators :@address, :ipv4?, :ipv6?
+
+    class << self
+      def from_string(str)
+        address, prefix = str.split("/")
+        prefix = prefix.to_i if prefix
+        new(address, prefix)
+      end
+    end
+
+    # @return [Integer] IPv4 address default prefix
+    IPV4_DEFAULT_PREFIX = 32
+    # @return [Integer] IPv6 address default prefix
+    IPV6_DEFAULT_PREFIX = 128
+
+    # Constructor
+    #
+    # @param address [String] IP address without the prefix
+    # @param prefix [Integer] IP prefix (number of bits). If not specified, 32 will be used for IPv4
+    #   and 128 for IPv6.
+    def initialize(address, prefix = nil)
```

Comment:
so we drop netmask completelly?

Reply:
Yes, basically we keep the address and the prefix. The netmask, if needed, can be inferred from there. Take into account that this class exists only to workaround the original IPAddr limitations.